src/ApiService/ApiService/OneFuzzTypes/Validated.cs

```diff
@@ -1,161 +1,107 @@
-﻿using System.Diagnostics;
-using System.Diagnostics.CodeAnalysis;
+﻿using System.Diagnostics.CodeAnalysis;
 using System.Text.Json;
 using System.Text.Json.Serialization;
 using System.Text.RegularExpressions;
 using Azure.Core;
 
 namespace Microsoft.OneFuzz.Service;
 
-static class Check {
-    private static readonly Regex _isAlnum = new(@"\A[a-zA-Z0-9]+\z", RegexOptions.Compiled);
-    public static bool IsAlnum(string input) => _isAlnum.IsMatch(input);
+static partial class Check {
+    [GeneratedRegex("\\A[a-zA-Z0-9]+\\z")]
```

Comment:
Interesting read about `GeneratedRegex` for anyone that's reviewing this:

https://devblogs.microsoft.com/dotnet/regular-expression-improvements-in-dotnet-7/#source-generation

Reply:
The C# linter suggested doing this so I let it make the change at the same time I was reworking the code.